tests/bsim/bluetooth/mesh/src/test_beacon.c

```diff
@@ -908,6 +971,539 @@ static void test_rx_secure_beacon_interval(void)
+}
+
+static void test_rx_priv_on_iv_update(void)
+{
+	struct bt_mesh_priv_beacon val;
+	int err;
+
+	common_rx_priv_setup();
+
+	val.enabled = 1;
+	val.rand_interval = 1;
+
+	err = bt_mesh_priv_beacon_cli_set(0, TX_ADDR, &val);
+	if (err) {
+		FAIL("Failed to enable Private Beacon (err=%d)", err);
+		return;
+	}
+
+	common_test_priv_rx_ivu();
+
+	PASS();
+}
+
+static void test_rx_priv_on_iv_update_long_int(void)
+{
+	struct bt_mesh_priv_beacon val;
+	int err;
+
+	common_rx_priv_setup();
+
+	val.enabled = 1;
+	val.rand_interval = 3;
+
+	err = bt_mesh_priv_beacon_cli_set(0, TX_ADDR, &val);
+	if (err) {
+		FAIL("Failed to enable Private Beacon (err=%d)", err);
+		return;
+	}
+
+	common_test_priv_rx_ivu();
+
+	PASS();
+}
+
+static void test_tx_priv_on_key_refresh(void)
+{
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &tx_comp);
+	test_priv_tx_init();
+
+	tx_on_key_refresh_test();
+}
+
+static void common_test_rx_priv_on_key_refresh(void)
+{
+	const uint8_t new_key[16] = {0x01};
+	uint8_t phase;
+	uint8_t status;
+
+	/* shift beaconing time line to avoid boundary cases. */
+	k_sleep(K_SECONDS(1));
+
+	status = bt_mesh_subnet_kr_phase_get(BT_MESH_KEY_PRIMARY, &phase);
+	ASSERT_TRUE(status == STATUS_SUCCESS);
+	ASSERT_TRUE(phase == BT_MESH_KR_NORMAL);
+
+	/* shift beaconing time line to avoid boundary cases. */
+	k_sleep(K_SECONDS(1));
+
+	status = bt_mesh_subnet_update(BT_MESH_KEY_PRIMARY, new_key);
+	ASSERT_TRUE(status == STATUS_SUCCESS);
+	status = bt_mesh_subnet_kr_phase_get(BT_MESH_KEY_PRIMARY, &phase);
+	ASSERT_TRUE(status == STATUS_SUCCESS);
+	ASSERT_TRUE(phase == BT_MESH_KR_PHASE_1);
+
+	wait_for_next_priv_beacon(BEACON_INTERVAL);
+
+	status = bt_mesh_subnet_kr_phase_get(BT_MESH_KEY_PRIMARY, &phase);
+	ASSERT_TRUE(status == STATUS_SUCCESS);
+	ASSERT_TRUE(phase == BT_MESH_KR_PHASE_1);
+
+	ASSERT_TRUE(memcmp(last_random, curr_random, 13) != 0);
+
+	memcpy(last_random, curr_random, 13);
+
+	wait_for_next_priv_beacon(BEACON_INTERVAL);
+
+	status = bt_mesh_subnet_kr_phase_get(BT_MESH_KEY_PRIMARY, &phase);
+	ASSERT_TRUE(status == STATUS_SUCCESS);
+	ASSERT_TRUE(phase == BT_MESH_KR_PHASE_2);
+
+	ASSERT_TRUE(memcmp(last_random, curr_random, 13) != 0);
+
+	memcpy(last_random, curr_random, 13);
+
+	wait_for_next_priv_beacon(BEACON_INTERVAL);
+
+	status = bt_mesh_subnet_kr_phase_get(BT_MESH_KEY_PRIMARY, &phase);
+	ASSERT_TRUE(status == STATUS_SUCCESS);
+	ASSERT_TRUE(phase == BT_MESH_KR_NORMAL);
+}
+
+static void test_rx_priv_on_key_refresh(void)
+{
+	struct bt_mesh_priv_beacon val;
+	int err;
+
+	common_rx_priv_setup();
+
+	val.enabled = 1;
+	val.rand_interval = 1;
+
+	err = bt_mesh_priv_beacon_cli_set(0, TX_ADDR, &val);
+	if (err) {
+		FAIL("Failed to enable Private Beacon (err=%d)", err);
+		return;
+	}
+
+	common_test_rx_priv_on_key_refresh();
+
+	PASS();
+}
+
+static void test_rx_priv_on_key_refresh_long_int(void)
+{
+	struct bt_mesh_priv_beacon val;
+	int err;
+
+	val.enabled = 1;
+	val.rand_interval = 3;
```

Comment:
The same here. Only this line is diff from the previous function.

Reply:
this is now passed as argument, for bot IVU and KR tests